llm-translator.html

```diff
@@ -0,0 +1,191 @@
+          url: ROOM_URL,
+          //startAudioOff: true,
+          //startVideoOff: true,
+          subscribeToTracksAutomatically: false,
+          audioSource: false,
+          videoSource: false,
+          dailyConfig: {},
+        });
+
+        [
+          "participant-updated",
+          "participant-left",
+          "error",
+          "camera-error",
+          "started-camera",
+          "active-speaker-change",
+          "recording-started",
+          "recording-stopped",
+          "recording-error",
+          "track-started",
+          "track-stopped",
+        ].forEach((e) => {
+          call.on(e, (ev) => {
+            console.log(`🔥 ${e}:`, ev);
+          });
+        });
+
+        call.on("track-started", startTrack);
+        call.on("track-stopped", stopTrack);
+
+        call.on("participant-joined", (ev) => {
+          console.log("🤠 Participant joined: ", ev);
+          const p = ev.participant;
+          // is this a person or a translator bot?
+          if (isTranslator(p)) {
+            // This is a translator. Update the list of languages
+            // to include this translator's language
+            console.log("this is a translator");
+            const lang = translatorLanguage(p);
+            const newOpt = document.createElement("option");
+            newOpt.value = lang;
+            newOpt.text = cap(lang);
+            languageSelect.appendChild(newOpt);
+          } else {
+            // This is a human. Subscribe to their video, and their
+            // audio if the current language is English
+            console.log("this is a human");
+            console.log(languageSelect.value);
+            const audioSub = languageSelect.value === "english";
+            call.updateParticipant(p["session_id"], {
+              setSubscribedTracks: {
+                video: true,
+                audio: audioSub,
+              },
+            });
+          }
+        });
+
+        call.on("app-message", (e) => {
+          console.log("Got app message: ", e);
+          const lang = languageSelect.value;
+          if (e.fromId === "transcription" && lang === "english") {
+            console.log("got it for english: ", e.data.text);
+            showSubtitle(e.data.session_id, e.data.text);
+          } else if (e.data.translation_language === lang) {
+            console.log("got it for non-english: ", e.data.translation);
+            showSubtitle(e.data.session_id, e.data.translation);
+          }
+          e.fromId;
+        });
+      }
+
+      function startTrack(evt) {
+        console.log("track started: ", evt);
+        if (evt.track.kind === "audio" && !evt.participant.local) {
+          let audiosDiv = document.getElementById("audios");
+          let audioEl = document.createElement("audio");
+          audiosDiv.appendChild(audioEl);
+          audioEl.style.width = "100%";
+          audioEl.srcObject = new MediaStream([evt.track]);
```

Comment:
I am doing a pretty quick read-through and not yet sure if call controls are included here. If the track can only start once (or maybe on meeting move), this probably works. But if we want to handle people starting/stopping their devices it may be nicer to implement conditional track replacement vs destroying and creating media streams each time. E.g., just replacing a track if it's changed: https://github.com/daily-demos/daily-samples-js/blob/main/samples/client-sdk/custom-session-data/src/call/dom.js#L22